feature-libs/checkout/base/occ/adapters/occ-checkout-delivery-modes.adapter.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  CheckoutDeliveryModesAdapter,
+  DELIVERY_MODE_NORMALIZER,
+} from '@spartacus/checkout/base/core';
+import {
+  ConverterService,
+  DeliveryMode,
+  normalizeHttpError,
+  Occ,
+  OccEndpointsService,
+} from '@spartacus/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError, map, pluck } from 'rxjs/operators';
+
+@Injectable()
+export class OccCheckoutDeliveryModesAdapter
+  implements CheckoutDeliveryModesAdapter
+{
+  constructor(
+    protected http: HttpClient,
+    protected occEndpoints: OccEndpointsService,
+    protected converter: ConverterService
+  ) {}
+
+  public getMode(userId: string, cartId: string): Observable<any> {
```

Comment:
As mentioned previously, I believe we can safely remove this

Reply:
let's add catchError((error) => throwError(normalizeHttpError(error))), if somehow we decide to keep

Reply:
Removed

Reply:
don't remove! :d since we kept the abstract method in the abstract class

Reply:
https://github.com/SAP/spartacus/pull/14165#discussion_r752114130

Reply:
brought it back (because it still existed in the spec too* :P)